### DISCUSSION TITLE
[feature req] convert pictures, allow for options

### DISCUSSION THREAD

**Discussion by jficz** [cwilby/nextcloud-workflow-media-converter#427] [Ideas]
As a user and/or admin I'd like to be able to automatically convert incoming pictures in various formats to a "normalized" format which I'd define.

Not a problem to have multiple flows for different input formats.

The output format should have at least these options:
 - "envelope" or a maximum picture size (i.e. scale everything down to a specified maximum size, like 5MPx or 3200x3200 or similar, keep aspect ratio, do not upscale)
 - convert to a specified format (AVIF, HEIFF, WEBP, JP2K, ...)
 - set desired target quality
 
 or set specific cmdline for imagick or whatever tool is going to be used

**Comment by cwilby** [comment]
I'm happy for somebody else to build/own this feature, but not ready to build it / own it myself yet.
